packages/playground/website/builder/builder.js

```diff
@@ -0,0 +1,917 @@
+			editor
+				.getSession()
+				.insert(
+					{ row: event.end.row, column: event.start.column + 1 },
+					']'
+				);
+			return;
+		} else if (inserted === '{') {
+			editor
+				.getSession()
+				.insert(
+					{ row: event.end.row, column: event.start.column + 1 },
+					'}'
+				);
+			return;
+		} else if (inserted === ',') {
+			const nextIndent = (lines[1 + event.start.row].match(/^(\s+)/g) || [
+				'',
+			])[0];
+			if (nextIndent.length >= indent.length) {
+				return;
+			}
+			if (lines[event.start.row][-1 + event.start.column] !== '"') {
+				editor
+					.getSession()
+					.insert(
+						{ row: event.end.row, column: event.end.column },
+						'\n' + indent
+					);
+				editor.moveCursorTo(1 + event.end.row, 1 + indent.length);
+				return;
+			}
+			editor
+				.getSession()
+				.insert(
+					{ row: event.end.row, column: event.end.column },
+					'\n' + indent + '""'
+				);
+			editor.moveCursorTo(1 + event.end.row, 1 + indent.length);
+			editor.execCommand('startAutocomplete');
+		}
+	});
+
+	window.test = {
+		iframeSrc,
+		iframe,
+		textarea,
+		button,
+	};
+
+	button.addEventListener('click', () => {
+		try {
+			clearError();
+			runBlueprint(editor);
+		} catch (error) {
+			showError(error);
+		}
+	});
+
+	let prevWin;
+
+	newTab.addEventListener('click', () => {
+		runBlueprint(editor);
+		const query = new URLSearchParams();
+
+		query.set('mode', 'seamless');
+		const url =
+			`https://playground.wordpress.net/?${query}#` +
+			JSON.stringify(getCurrentBlueprint(editor));
+		if (prevWin) {
+			prevWin.close();
+		}
+		prevWin = window.open(url, '_blank');
+	});
+
+	if (window.location.hash) {
+		loadFromHash(editor);
+	} else {
+		formatJson(editor, {
+			landingPage: '/wp-admin/',
+			phpExtensionBundles: ['kitchen-sink'],
+			preferredVersions: {
+				php: '7.4',
+				wp: '5.9',
+			},
+			steps: [
+				{
+					step: 'login',
+					username: 'admin',
+					password: 'password',
+				},
+			],
+		});
+	}
+
+	runBlueprint(editor);
+
+	window.addEventListener('hashchange', () => {
+		loadFromHash(editor);
+		runBlueprint(editor);
+	});
+
+	editor.commands.addCommand({
+		name: 'Run Blueprint',
+		bindKey: {
+			win: 'Ctrl-Enter|Ctrl-S',
+			mac: 'Command-Enter|Command-S',
+		},
+		exec: (editor) => runBlueprint(editor),
+		readOnly: false,
+	});
+
+	const save = document.querySelector('button#save');
+	const open = document.querySelector('button#open');
+
+	const saveMethod = () => {
+		const dataUri = `data:application/json;base64,${btoa(
+			JSON.stringify(JSON.parse(editor.getValue()), null, 2)
+		)}`;
+		const link = document.createElement('a');
+		link.setAttribute('href', dataUri);
+		link.setAttribute(
+			'download',
+			`blueprint-${new Date().toISOString()}.json`
+		);
+		link.click();
+	};
+
+	const openMethod = () => {
+		const input = document.createElement('input');
+		input.setAttribute('type', 'file');
+		input.addEventListener('change', (event) => {
+			[...input.files].forEach((f) => {
```

Comment:
Let's not block this PR on code style changes anymore, but like @dmsnell and I mentioned in a few earlier rounds of feedback, let's avoid that destructuring syntax in general in favor of more readable patterns. In this case, this could be just `input.files`.